core/src/main/java/io/grpc/internal/ProxyDetectorImpl.java

```diff
@@ -258,12 +260,15 @@ private ProxyParameters detectProxy(InetSocketAddress targetAddr) throws IOExcep
     }
 
     if (auth == null) {
-      return new ProxyParameters(resolvedProxyAddr, null, null);
+      return ProxyParameters.forAddress(resolvedProxyAddr).build();
     }
 
     // TODO(spencerfang): users of ProxyParameters should clear the password when done
```

Comment:
I changed `ProxyParameters` to use [`char[]` for passwords](https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords) for this reason. 
What remains now in this direction is propagating this change to the Netty and OKHttp transports (the first one seems to expose a `String` password parameter in its public API)